### DISCUSSION TITLE
dear sir, a stupid question, how to compile on linux(debian/ ubuntu)?

### DISCUSSION THREAD

**Discussion by ericjing83** [uNetworking/uWebSockets#1727] [Q&A]
dear sir, 

a stupid question, how to compile on linux(debian/ ubuntu)?  I try "cmake" and "make", but I failed.
Could you give me some tips? e.g, some bash command? Thanks.

**Comment by uNetworkingAB** [comment]
It should be "make". So if that doesn't work you probably didnt clone recursively:

git clone --recursive https://github.com/uNetworking/uWebSockets

**Comment by uNetworkingAB** [comment]
(because of submodules)

**Comment by ericjing83** [comment]
thanks. by the way, does the latest version requires compiler with C++20 support? Is gcc 10 ok?

**Comment by uNetworkingAB** [comment]
C++17 is fine but designated initializers are needed but those have been in C for ages so practically all C++17 compilers work. But yes, kind of, to be strictly technical C++20 is needed
